### DISCUSSION TITLE
Are we only allowed one session per pairing phrase? Is it a litd or a mailbox limitation?

### DISCUSSION THREAD

**Discussion by huumn** [lightninglabs/lightning-terminal#751] [Q&A] [answered]
When testing our LNC implementation, I noticed only one tab could connect via LNC at a time. 

Is that right? Or did I misinterpret connection errors?

If that's right, and only one browser tab or device can consume an LNC session, how might I change that? Would I need to run a mailbox myself to increase the connection limit? Is it a litd configuration thing?

**Comment by guggero** [comment, marked as answer]
Yes, it's correct, you can only use a pairing phrase a single time. A pairing phrase corresponds to a single LNC session, which is unique. But you can create as many sessions (each with a different pairing phrase) with the same permissions (and optionally restricted to the same account ID) as you want and use them in parallel.

The restriction of single-use-pairing-phrase comes from the underlying cryptography protocol (SPAKE2+Noise) and cannot easily be changed without security tradeoffs.

**Reply by huumn** [reply]
~~I guess in a browser we'll need to use something like a web worker to share a session. Thanks!~~

If anyone comes across this, this isn't much of a problem really. 

The solution is to connect, perform the rpc call, then disconnect. For a single person, with multiple tabs or devices, there's no real contention over the session. It adds some latency but it's better than having only one tab work at a time or over-engineering with web workers.

**Reply by huumn** [reply]
~~Update update the above suggestion doesn't work as well as it initially appeared. Nearly all of my disconnections result in the following error on reconnects:~~

```console
Got failure on send socket/stream, re-trying: failed to write: WebSocket closed: unclean connection close: status = StatusAbnormalClosure and reason = ""
```

~~Initially I thought it was a race condition (reconnecting while in the middle of disconnecting), but even after waiting for `lndConn = nil` (ie `!isConnected` in `lnc-web`) to reconnect again, this still happens.~~

~~The only thing I haven't tried yet is replacing the `DialContext` with a fresh one by creating a new `LNC` instance when connecting.~~

-----------

~~Anyway, the details here probably belong in the lnc repo, but better here than nowhere.~~

Disregard above. `disconnect` in `lnc-web` opaquely clears decrypted credentials so I was closing an unopened connection which resulted in the above errors.
